plugins/inputs/nats_consumer/nats_consumer.go

```diff
@@ -125,7 +131,7 @@ func (n *NatsConsumer) Start(acc telegraf.Accumulator) error {
 		// Setup message and error channels
 		n.errs = make(chan error)
 
-		n.in = make(chan *nats.Msg, 1000)
+		n.in = make(chan *nats.Msg, n.PendingMessageLimit)
```

Comment:
Aren't the `PendingMessageLimit` and the `MaxUndeliveredMessages` somehow the same thing? Just wondering...